kokoro/scripts/test/go_test.sh

```diff
@@ -193,11 +193,6 @@ if [[ "${TEST_SUITE_NAME}" == "os_config_test" ]]; then
 fi
 
 STDERR_STDOUT_FILE="${KOKORO_ARTIFACTS_DIR}/test_stderr_stdout.txt"
-function produce_xml() {
-  cat "${STDERR_STDOUT_FILE}" | "$(go env GOPATH)/bin/go-junit-report" > "${LOGS_DIR}/sponge_log.xml"
-}
-# Always run produce_xml on exit, whether the test passes or fails.
-trap produce_xml EXIT
```

Comment:
This part might still be relevant, right? If the test crashes/panics, shouldn't we still get an XML file?

Reply:
As long as the `gotestsum` doesn't crash, we still will. If that crashes, then we should be looking in the `go_test` log.

Reply:
I would confirm with @martijnvans — I seem to recall that this trap was originally added because otherwise we ended up with radio silence from CI runs where the test runner crashed. But maybe it's no longer needed?

Reply:
I believe that if the test panics, we won't get an xml file (before this CL). Afterwards? Anyone's guess. We can try adding a panic and see how `gotestsum` reacts. I bet it does something reasonable though.

Reply:
Since the purpose of this PR just swapping to `gotestsum`, and it sounds like this case is being handled the same as before. I'll close this.